nextcord/state.py

```diff
@@ -2259,6 +2264,60 @@ def parse_guild_scheduled_event_user_remove(self, data) -> None:
                 data["guild_id"],
             )
 
+    def add_automod_rule(self, data: AutoModerationRulePayload) -> AutoModerationRule:
+        rule = AutoModerationRule(state=self, guild=self._get_guild(int(data["guild_id"])), data=data)  # type: ignore -- automod rules can't be created in non-guilds
+        self._automod_rules[int(data["id"])] = rule
+        return rule
+
+    def delete_automod_rule(self, id: int) -> None:
+        self._automod_rules.pop(id, None)
+
+    def get_automod_rule(self, id: int) -> Optional[AutoModerationRule]:
+        return self._automod_rules.get(id)
+
+    def update_automod_rule(self, data: AutoModerationRulePayload) -> None:
+        self.delete_automod_rule(int(data["id"]))
+        self.add_automod_rule(data)
+
+    def parse_auto_moderation_rule_create(self, data: AutoModerationRulePayload) -> None:
+        self.add_automod_rule(data)
+        self.dispatch(
+            "automod_rule_create",
+            AutoModerationRule(state=self, guild=self._get_guild(int(data["guild_id"])), data=data),  # type: ignore
+        )
+
+    def parse_auto_moderation_rule_update(self, data: AutoModerationRulePayload) -> None:
+        old_data = self.get_automod_rule(int(data["id"]))
+        self.update_automod_rule(data)
+        self.dispatch(
+            "automod_rule_update",
+            old_data,
+            AutoModerationRule(state=self, guild=self._get_guild(int(data["guild_id"])), data=data),  # type: ignore
+        )
+
+    def parse_auto_moderation_rule_delete(self, data: AutoModerationRulePayload) -> None:
+        self.delete_automod_rule(int(data["id"]))
+        self.dispatch(
+            "automod_rule_delete",
+            AutoModerationRule(state=self, guild=self._get_guild(int(data["guild_id"])), data=data),  # type: ignore
+        )
+
+    def parse_auto_moderation_action_execution(
+        self, data: AutoModerationActionExecutedEvent
+    ) -> None:
+        self.dispatch("automod_action_execution", AutoModerationAction(data["action"]))
+
+    async def _add_automod_rule_from_guild_data(self, data: GuildPayload):
+        id = int(data["id"])
+        try:
+            rules = await self.http.list_guild_automod_rules(guild_id=id)
+            for rule in rules:
+                self.add_automod_rule(data=rule)
+        except Forbidden:
+            pass
+        except NotFound:  # idk why it raises but h
```

Comment:
Remove this comment, or at-least make it more suitable for the code base your contributing to

Reply:
```suggestion
        except NotFound:
```